metrics/metrics-core/src/test/java/io/servicecomb/metrics/core/TestEventAndRunner.java

```diff
@@ -0,0 +1,94 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicecomb.metrics.core;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import io.servicecomb.core.metrics.InvocationFinishedEvent;
+import io.servicecomb.core.metrics.InvocationStartProcessingEvent;
+import io.servicecomb.core.metrics.InvocationStartedEvent;
+import io.servicecomb.foundation.common.utils.EventUtils;
+import io.servicecomb.metrics.core.event.DefaultEventListenerManager;
+import io.servicecomb.metrics.core.model.RegistryModel;
+import io.servicecomb.metrics.core.schedule.DefaultStatisticsRunner;
+import io.servicecomb.swagger.invocation.InvocationType;
+
+public class TestEventAndRunner {
+
+  @Test
+  public void test() throws InterruptedException {
+
+    DefaultEventListenerManager manager = new DefaultEventListenerManager();
+
+    EventUtils.triggerEvent(new InvocationStartedEvent("fun1", System.nanoTime()));
+    EventUtils.triggerEvent(
+        new InvocationStartProcessingEvent("fun1", InvocationType.PRODUCER, System.nanoTime(),
+            TimeUnit.MILLISECONDS.toNanos(100)));
+    EventUtils
+        .triggerEvent(new InvocationFinishedEvent("fun1", InvocationType.PRODUCER, System.nanoTime(),
+            TimeUnit.MILLISECONDS.toNanos(200), TimeUnit.MILLISECONDS.toNanos(300)));
+
+    EventUtils.triggerEvent(new InvocationStartedEvent("fun1", System.nanoTime()));
+    EventUtils.triggerEvent(
+        new InvocationStartProcessingEvent("fun1", InvocationType.PRODUCER, System.nanoTime(),
+            TimeUnit.MILLISECONDS.toNanos(300)));
+    EventUtils
+        .triggerEvent(new InvocationFinishedEvent("fun1", InvocationType.PRODUCER, System.nanoTime(),
+            TimeUnit.MILLISECONDS.toNanos(400), TimeUnit.MILLISECONDS.toNanos(700)));
+
+    EventUtils.triggerEvent(new InvocationStartedEvent("fun12", System.nanoTime()));
+    EventUtils.triggerEvent(
+        new InvocationStartProcessingEvent("fun12", InvocationType.PRODUCER, System.nanoTime(),
+            TimeUnit.MILLISECONDS.toNanos(500)));
+    EventUtils
+        .triggerEvent(new InvocationFinishedEvent("fun12", InvocationType.PRODUCER, System.nanoTime(),
+            TimeUnit.MILLISECONDS.toNanos(600), TimeUnit.MILLISECONDS.toNanos(1100)));
+
+    EventUtils.triggerEvent(new InvocationStartedEvent("fun11", System.nanoTime()));
+
+    DefaultStatisticsRunner runner = new DefaultStatisticsRunner();
+
+    //we need some time escape for sim statistics interval
+    Thread.sleep(100);
+    runner.run();
+
+    RegistryModel model = runner.getRegistryModel();
+
+    Assert.assertTrue(model.getInvocations().get("fun1").getCountInQueue() == 0);
```

Comment:
Please use assertEquals instead of assertTrue to help the user to find out the reason of the test failure.

Reply:
Done